### PR TITLE
Add class GlobalWellInfo and use it from WellState

### PR DESCRIPTION
**Pull request by joakim-hove**
Create a small class with global well information in order to extract "stuff" from the WellStateFullyImplicitBlackOil class.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Comments to this?

File: opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp [hunk 1]
```diff
@@ -36,6 +37,7 @@
 #include <functional>
 #include <iostream>
 #include <map>
+#include <optional>
```
Comment (bska) [inline opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp:40]: Could you add this header in alphabetic order (after `<numeric>`)?

File: opm/simulators/wells/GlobalWellInfo.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+std::size_t GlobalWellInfo::well_index(const std::string& wname) const {
+    return this->name_map.at(wname);
+}
+
+const std::string& GlobalWellInfo::well_name(std::size_t well_index) const {
+    for (const auto& [name, index] : this->name_map) {
+        if (index == well_index)
+            return name;
+    }
+    throw std::logic_error("No well with index: " + std::to_string(well_index));
```
Comment (bska) [inline opm/simulators/wells/GlobalWellInfo.cpp:86]: You need to include `<stdexcept>` to use `std::logic_error`.

**Review by bska — COMMENTED** [review]
Looks good for the most part.  I have a couple of minor comments, but otherwise I think this should be merged.

That said, I always get confused by `Comm::sum()`'s behaviour so we could possibly add a comment to `GlobalWellInfo::communicate()` that `Comm::sum()` is an "all to all" operation that accumulates corresponding array elements and distributes the resulting array to all processes.

**Comment by joakim-hove** [comment]
OK - I have updated.

**Comment by joakim-hove** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks for the updates.  I'll merge this into master now.

**Pull request merged**